YuMi ROS integration

FWachter opened the pull request:
Added YuMi files to work with the real robot and attached the camera assembly to the left hand. Also created a file to automatically set up the workspace on a new desktop.

FWachter committed:
Added Camera Assembly to YuMi model

Added camera assembly to YuMi model, along with other configuration files to run YuMi scripts as command line inputs

FWachter committed:
Removed unnecessary comments in URDF

Remove the safety controller commented sections from the YuMi URDF

FWachter committed:
Added file to setup WS for new users

Added file that will setup the correct workspace in order to run the YuMi files. This will be used on desktops that have not been setup yet to run the YuMi files.

FWachter committed:
Modified setup file

Did some small modifications to the setup file. It is still incomplete though.

FWachter committed:
Removed error causing joint publisher to die

Joint state publisher would die due to commenting method used. Adjusted the comments to stop this process from dying.

FWachter committed:
Merge branch 'master' of https://github.com/ethz-asl/yumi into feature/CameraArm

Updating files from master

FWachter committed:
Adjusted .gitignore file

Allowed for URDF's, DAE, and STL files to be added, and removed OSX generated files and others

FWachter committed:
Modified setup script

Made final modifications before testing in new linux machine to find errors

FWachter committed:
Added to setup workspace file

Needs to be tested, should be able to build the full workspace for a new user.

FWachter committed:
Made final changes to workspace setup script

Tested this script on a new desktop, made final adjustments

FWachter committed:
Made final modification to workspace setup script

Tested the final version of the setup script and applied final changes

FWachter committed:
Modified setup workspace file

Added prompts for adding in terminal commands and sourcing, removed file in yumi_scripts folder (addYuMiCommands.bash) and placed code into setupWSVariables.bash to make the setup more staightforward with less files.

FWachter committed:
Tested and fixed typo's

Fixed typo's and adjusted formatting of prompts

ffurrer commented on yumi_description/urdf/yumi.urdf:
```diff
@@ -0,0 +1,956 @@
+<!-- =================================================================================== -->
+<!-- |    This document was autogenerated by xacro from yumi.urdf.xacro                | -->
+<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
+<!-- =================================================================================== -->
```
I think the general idea here is, that you edit `yumi.urdf.xacro` instead of this file. And try to keep the values you entered as parameters, this makes it easier to modify them. (Just some params in the beginning of the file).

ffurrer commented on yumi_description/meshes/base/README.txt:
```diff
@@ -10,27 +10,27 @@ Work Around:
+<file_name> - string - name of the STL file
+<origin_offset> - "dx dy dz" - offset retrieved from SolidWorks, replace dx, dy, and dx appropriately
 
 
```
+1 for updating this :)

ffurrer commented:
Just one comment, consider editing the `xacro` file instead of the `urdf`. Otherwise looks good to merge.

FWachter committed:
Updated YuMi URDF/XACRO files

Remove unnecessary XACRO files, and create new files in order to generate the proper YuMi URDF file

FWachter commented:
Create the `xacro` files to generate the `urdf`. Do you want to look over this or should I go ahead and merge the pull request?

ffurrer commented on yumi_description/launch/yumi.rviz:
```diff
@@ -1,236 +0,0 @@
-Panels:
```
why did you delete the rviz configuration? I think in general it's nice to have one.

FWachter replied:
They were more or less the same thing as this [rviz file](https://github.com/ethz-asl/yumi/blob/master/yumi_moveit_config/launch/moveit.rviz) and this [launch file](https://github.com/ethz-asl/yumi/blob/master/yumi_moveit_config/launch/moveit_planning_execution.launch). The RViz file that was deleted wan't updated to include the camera assembly. Should I put them back or are the linked ones good enough?

ffurrer replied:
if they are the same, yes, then there is no need to add new ones ;) But of course it's nice to provide a proper customized `.rviz` file, but I guess, that's fine for now.

ffurrer commented on yumi_description/launch/yumi_rviz.launch:
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0"?>
```
same

ffurrer commented on yumi_description/urdf/materials.xacro:
```diff
@@ -41,4 +49,18 @@
     <color rgba="1.0 1.0 1.0 1.0"/>
   </material>
 
+  <material name="Drexel_Blue">
```
drexel blue? ;)

ffurrer commented on yumi_description/urdf/yumi.urdf:
```diff
@@ -598,13 +520,9 @@
   </link>
-  <!--Extensions -->
-  <!-- body -->
-  <gazebo reference="yumi_body">
```
if you remove these, there will be no colors in gazebo, I guess.

FWachter replied:
Ya, I'm not sure if it affected anything. I tried running it with that code in there and without, and it seems as if it didn't matter as long as whichever color I gave it was in the [materials.xacro](https://github.com/ethz-asl/yumi/blob/addition/SetupWS/yumi_description/urdf/materials.xacro) file.

ffurrer replied:
yeah the material stuff is used in _rviz_ and the `<gazebo ...>` stuff is used in _gazebo_. (you have to have both ;))

ffurrer commented:
I think the `yumi_description/urdf/yumi.urdf` file should be removed?

FWachter commented:
Is there a reason for not including the URDF? I noticed other repo's don't include it, but wasn't really sure why. Is it to make sure that whoever clones the repo generates the most updated file or some other reason?

ffurrer commented:
The reason is basically, that you don't add files to a repo that in general have no implications if you change them, at least they shouldn't, cause all the changes should be done in the xacro files, and yes, then it's clear that the urdf file is a generated file. (also if you remove the urdf file and it still works, you will figure out if it also works for someone else ;))

FWachter committed:
Updated XACRO to include Gazebo elements, and updated planning context

Gazebo elements were put back into the XACRO files, removed URDF for YuMi, and updated the planning context launch file to launch the yumi.urdf.xacro file instead of a premade URDF file

FWachter commented:
Made the following changes: Added back in all the Gazebo elements, removed YuMi `urdf`, and updated the planning context launch file to launch from the `xacro` file (didn't know that was possible before). I will update the RViz file that was deleted previously in the coming days. Any other suggestions?

ffurrer commented:
lgtm
